module_utils/network_lsr/argument_validator.py

```diff
@@ -1611,7 +1663,9 @@ def _validate_post(self, value, name, result):
                         "invalid 'interface_name' '%s'" % (result["interface_name"]),
                     )
             else:
-                if not result.get("mac"):
+                if not result.get("mac") and (
+                    not result.get("match") or not result.get("match", {}).get("path")
+                ):
```

Comment:
The reason that I written this way is because we may want to add kernel-command-line or interface_name into the match section. (Maybe it is only my assumption) 
 
 I do not see a possibility where `result.get("match", {}).get("path")` is not None, but `result.get("match")` is.

Reply:
when match setting is defined as `match: None`, `not result.get("match")` is necessary, otherwise,   `not result.get("match", {}).get("path")` will raise an error -> `'NoneType' object has no attribute 'get'`

Reply:
I see, yes, you are right. Looking back at the code it seems to me that `match: None` would be normalized to `match: {}` now, though. Wouldn't it?

Reply:
Updated. I do not see a possibility where `result.get("match", {}).get("path")` is not None, but `result.get("match")` is. Because the default value for match setting is None and of course user can also specify a match setting. So there is always a key "match" in the connection dictionary (`"match" in result is True`).

Reply:
>I see, yes, you are right. Looking back at the code it seems to me that `match: None` would be normalized to `match: {}` >now, though. Wouldn't it?

If user does not clearly specify `match: None`,  then default value for match setting (`match:None`) will be preserved.   If user clearly specify `match: None`, then it will be normalized into `match: {}`.

Reply:
To me it seems to make more sense to normalize `match: None` and match not being specified both to the same, in this case `match: {}`. This means that no match setting will be added to the profile/an existing one will be removed, doesn't it?

Reply:
Ok, I changed the default value for match setting into `match: {}`